src/scripts/main.js

```diff
@@ -1 +1,19 @@
 'use strict';
+
+document.addEventListener('DOMContentLoaded', () => {
+  const mainImage = document.querySelector('#largeImg');
+
+  const thumbnails = document.querySelector('#thumbs');
+
+  function changeImage(e) {
+    if (e.target.tagName === 'IMG' || e.target.tagName === 'A') {
+      e.preventDefault();
+
+      const necessaryLink = e.target.closest('a');
+      const newSrc = necessaryLink.getAttribute('href');
+
+      mainImage.setAttribute('src', newSrc);
+    }
```

Comment:
In the if condition, you are checking whether the clicked element is either an image or a link. However, according to your code, the `changeImage` function should only be triggered when the clicked element is an image inside a link. Therefore, you should adjust your condition to check whether the clicked element is an image and its closest parent is a link. This will ensure that the function only runs when necessary, improving the performance of your code.